File: Cabify/2021/problemH.go
package main

import "fmt"

func main() {
	var c int
	fmt.Scanf("%d", &c)
	n := make([]int, c)
	for i := range n {
		fmt.Scanf("%d", &n[i])
	}

	s := make(map[int]bool)

	for {
		found := false

		max := n[0]
		for i := 1; i < len(n); i++ {
			if max < n[i] {
				max = n[i]
			}
		}

		for v := 2; v <= max; v++ {
			allEq := true
			x := n[0] % v
			for _, j := range n {
				if j%v != x {
					allEq = false
					break
				}
			}

			if allEq {
				found = true
				news := make([]int, 0)
				for k := range s {
					news = append(news, k*v)
				}
				for _, h := range news {
					s[h] = true
				}
				s[v] = true

				for h := range n {
					n[h] /= v
				}
				break
			}
		}

		if !found {
			break
		}
	}
	f := true
	for k := range s {
		if !f {
			fmt.Print(" ")
		}
		fmt.Print(k)
		f = false
	}
	fmt.Println("")
}

File: Cabify/2021/problemH.py
c = int(input())
plates = []
for _ in range(c):
  plates.append(int(input()))

results = set()
while True:
  found = False
  for value in range(2, max(plates)):
    all_eq = True
    target = plates[0]%value
    for plate in plates:
      if plate%value != target:
        all_eq = False
        break
    
    if all_eq:
      found = True

      for v in results.copy():
        results.add(v*value)
      results.add(value)

      plates = list(map(lambda x: int(x/value), plates))

      break

  if not found:
    break

print(" ".join([str(x) for x in results]))